ttg/ttg/base/tt.h

```diff
@@ -100,29 +99,36 @@ namespace ttg {
     }
 
   private:
+   // non-copyable
    TTBase(const TTBase &) = delete;
    TTBase &operator=(const TTBase &) = delete;
-   TTBase(TTBase &&) = delete;
-   TTBase &operator=(TTBase &&) = delete;
+
+   // movable
+   TTBase(TTBase && other) : instance_id(other.instance_id)
```

Comment:
Is it intentional to make the move ctor/op private? Does that actually allow moving?